File: TD/TD2/Code/7.cpp
/**
 * Calculer les rastd::cines rééles d'un polymer du second degré
 */
#include <cmath> // Contient la fonction sqrt()
#include <iostream>

int main() 
{
  float a;
  float b;
  float c;
  float d;

  std::cout << "Donne moi trois chiffres : " << std::flush;
  std::cin >> a >> b >> c;

  d = b * b - 4 * a * c;

  if (d < 0)
    std::cout << "Pas de racines réelles" << std::endl;
  else if (d == 0)
    std::cout << "Une racine double : " << -b / (2 * a) << std::endl;
  else
    std::cout << (-b + sqrt(d)) / (2 * a) << " " 
        << (-b - sqrt(d)) / (2 * a) << std::endl;

  return 0;
}

File: TD/TD2/Code/Python/7.py
#!/usr/bin/python
# Calculer les racines rééles d'un polymer du second degré
from math import sqrt

print('Donne moi trois entiers, un par un : ', end='')
a = float(input())
b = float(input())
c = float(input())

d = b * b - 4 * a * c

if d < 0:
    print('Pas de racine réels.')
elif d == 0:
    print('Une racine double', -b / (2 * a))
else:
    print((-b + sqrt(d)) / (2 * a), (-b - sqrt(d)) / (2 * a))
